File: Server/DogMate/src/main/java/com/DogMate/util/PhoneValidation.java
package com.DogMate.util;

/**
 * Israeli mobile numbers: 10 digits starting with {@code 05}, or 9 digits starting with {@code 5}
 * (same national number without leading zero).
 */
public final class PhoneValidation {

    private PhoneValidation() {
    }

    /**
     * @return digits only (e.g. {@code 0501234567} or {@code 501234567})
     * @throws IllegalArgumentException if missing or not a valid Israeli mobile pattern
     */
    public static String requireValidIsraeliMobile(String raw) {
        if (raw == null || raw.isBlank()) {
            throw new IllegalArgumentException("נדרש מספר טלפון");
        }
        String digits = raw.replaceAll("\\D", "");
        if (digits.isEmpty()) {
            throw new IllegalArgumentException("נדרש מספר טלפון");
        }
        boolean validTen = digits.length() == 10 && digits.startsWith("05");
        boolean validNine = digits.length() == 9 && digits.charAt(0) == '5';
        if (!validTen && !validNine) {
            throw new IllegalArgumentException("מספר פלאפון ישראלי לא תקין");
        }
        return digits;
    }

    /**
     * For optional phone (e.g. dog owner registration): returns {@code null} if blank.
     * If non-blank, validates the same way as {@link #requireValidIsraeliMobile(String)}.
     */
    public static String optionalValidIsraeliMobile(String raw) {
        if (raw == null || raw.isBlank()) {
            return null;
        }
        return requireValidIsraeliMobile(raw);
    }
}

File: Client/DogMate/utils/phoneValidation.ts
/**
 * Israeli mobile: 10 digits starting with 05, or 9 digits starting with 5 (no leading 0).
 * Matches server {@link com.DogMate.util.PhoneValidation}.
 */

export function normalizeIsraeliMobileToDigits(
  raw: string | number | null | undefined
): string | null {
  if (raw == null || raw === '') return null;
  const digits = String(raw).replace(/\D/g, '');
  if (digits.length === 10 && digits.startsWith('05')) return digits;
  if (digits.length === 9 && digits.startsWith('5')) return digits;
  return null;
}

export function isValidIsraeliMobileInput(raw: string): boolean {
  return normalizeIsraeliMobileToDigits(raw) != null;
}

/** For tel: — digits only */
export function digitsForTelDial(raw: string | number | null | undefined): string | null {
  return normalizeIsraeliMobileToDigits(raw);
}

/**
 * Digits for WhatsApp `phone` query param: country code 972, no leading 0 (e.g. 972501234567).
 */
export function normalizeIsraeliMobileToWhatsAppPhoneParam(
  raw: string | number | null | undefined
): string | null {
  const d = normalizeIsraeliMobileToDigits(raw);
  if (!d) return null;
  if (d.length === 10 && d.startsWith('05')) return `972${d.slice(1)}`;
  if (d.length === 9 && d.startsWith('5')) return `972${d}`;
  return null;
}
